### ISSUE TITLE
Login via QR-Code? (when using iServ or Office365 login)

### ISSUE THREAD

**Issue by o0shojo0o** [JonasJoKuJonas/homeassistant-WebUntis#111]
Hi,
is it possible to record the login via QR code?

Further information:
https://github.com/python-webuntis/python-webuntis/issues/47

**Comment by o0shojo0o** [comment]
Kann ich weiter unterstützen um diesen Feature Request voranzutreiben?

**Comment by JonasJoKuJonas** [comment]
This feature has first to be addet to the [python-webuntis](https://github.com/python-webuntis/python-webuntis) reposetory

**Comment by henfri** [comment]
Hello Jonas,

thanks for providing homeassistantWebUntis!
In python-webuntis it will not be implemented, as it is not part of the official API.
However, as I understand, you only need the code from 
https://github.com/python-webuntis/python-webuntis/issues/47#issuecomment-1481971697
That should return you a session

If you are willing to include this, I could try to provide a working example.

Thanks and best regards,
Hendrik

**Comment by henfri** [comment]
@JonasJoKuJonas are you willing to integrate this?

**Comment by mrtncode** [comment]
@henfri @o0shojo0o @chgardener
We found a way to implement QR Code authentication. Is anyone still interested? Testers are welcome! The feature is expected to be released soon

**Comment by Panda1441** [comment]
I would be very interested in this feature because my school only allows the Office 365 login.

**Comment by mrtncode** [comment]
> I would be very interested in this feature because my school only allows the Office 365 login.

Nice! I will let you know when you can test it! Feedback is very appreciated

**Comment by mrtncode** [comment]
@chgardener @Panda1441 I merged the PR and released v3.1.0! I appreciate your feedback and hope it works for M365 login too :)

**Comment by Panda1441** [comment]
I don’t know if this is because my school year hasn’t started yet, but if I try to choose the data source of the timetable, I get this error message in the logs:

```
2026-08-27 19:08:22.665 ERROR (SyncWorker_1) [custom_components.webuntis.config_flow] Testing timetable for source: personal
2026-08-27 19:08:22.996 ERROR (SyncWorker_1) [webuntis] {'jsonrpc': '2.0', 'id': '2026-08-27 19:08:22.946645', 'error': {'message': 'startDate and endDate are not within a single school year.', 'code': -8507}}
2026-08-27 19:08:22.997 ERROR (SyncWorker_1) [custom_components.webuntis.config_flow] Error testing timetable: startDate and endDate are not within a single school year.
```

Do you have an idea where this is coming from?

**Comment by mrtncode** [comment]
@Panda1441 Yes, that's schoolyear related. It seems your school hasn't created the next school year so the integration is confused. Just wait until the school year starts and everything should work :)

**Comment by Panda1441** [comment]
Perfekt thanks for the help.

**Comment by chgardener** [comment]
I just used the login via qr code with the untis://.... link and it worked :) My school uses M365 btw.
